setup/setup-ci/security-scanner/amlsecscan.py

```diff
@@ -0,0 +1,759 @@
+        with open(_local_config_path, "rt") as file:
+            config.update(json.load(file))
+        _logger.debug(
+            f"logAnalyticsResourceId after loading config file: {config['logAnalyticsResourceId']}"
+        )
+
+    # Set Log Analytics workspace ARM Resource ID if passed via command-line parameter
+    if log_analytics_resource_id is not None:
+        config["logAnalyticsResourceId"] = log_analytics_resource_id
+        _logger.debug(
+            f"logAnalyticsResourceId after setting command-line parameter: {config['logAnalyticsResourceId']}"
+        )
+
+    # Retrieve Log Analytics workspace ARM Resource ID from Azure ML diagnostic settings if
+    # provided neither via local config file nor command-line parameter
+    if config.get("logAnalyticsResourceId", None) is None:
+        config["logAnalyticsResourceId"] = _get_log_analytics_from_diagnostic_settings()
+        _logger.debug(
+            f"logAnalyticsResourceId after querying Azure ML diagnostic settings: {config['logAnalyticsResourceId']}"
+        )
+
+    # Sanitize the Log Analytics workspace ARM Resource ID
+    config["logAnalyticsResourceId"] = _sanitize_log_analytics_resource_id(
+        config["logAnalyticsResourceId"]
+    )
+
+    _logger.debug(f"Configuration: {config}")
+
+    _logger.info(f"Writing configuration file {_global_config_path}")
+    with open(_global_config_path, "wt") as file:
+        json.dump(config, file, indent=2)
+    shutil.chown(_global_config_path, "azureuser", "azureuser")
+
+    _logger.info("Installing Trivy")
+    _run(
+        "apt-get install -y --no-install-recommends --quiet wget apt-transport-https gnupg lsb-release"
+    )
+    _run(
+        "wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | apt-key add -"
+    )
+    _run(
+        "echo deb https://aquasecurity.github.io/trivy-repo/deb $(lsb_release -sc) main | tee -a /etc/apt/sources.list.d/trivy.list"
+    )
+    _run("apt-get update")
+    _run("apt-get install -y --no-install-recommends --quiet trivy")
+
+    script_path = _config_folder_path + "/run.sh"
+    _logger.info(f"Writing script file {script_path}")
+    with open(script_path, "wt") as file:
+        file.write(
+            f"""#!/bin/bash
+set -e
+exec 1> >(logger -s -t AMLSECSCAN) 2>&1
+
+# Limit CPU usage to 20% and reduce priority (note: the configuration is not persisted during reboot)
+if [ ! -d /sys/fs/cgroup/cpu/amlsecscan ]
+then
+    mkdir -p /sys/fs/cgroup/cpu/amlsecscan
+    echo 100000 | tee /sys/fs/cgroup/cpu/amlsecscan/cpu.cfs_period_us > /dev/null
+    echo 20000 | tee /sys/fs/cgroup/cpu/amlsecscan/cpu.cfs_quota_us > /dev/null
+    echo 5 | tee /sys/fs/cgroup/cpu/amlsecscan/cpu.shares > /dev/null
+fi
+echo $$ | tee /sys/fs/cgroup/cpu/amlsecscan/tasks > /dev/null
+
+nice -n 19 python3 {os.path.abspath(__file__)} $1 $2 $3 $4 $5
+"""
+        )
+    os.chmod(script_path, 0o0755)
+
+    _logger.info(f"Writing crontab file /etc/cron.d/amlsecscan")
```

Comment:
Recommend using a systemd service/timer combo rather than cron for this. This allows you to set the start times more accurately (e.g. you can specify the behavior of a missed scan), and the logs go to journalctl instead of cron's log.

Reply:
Discussing with @deeikele offline, CRON jobs are sufficient to meet customer requirements. Next step will be to remove the need for this script by providing an integrated security solution.